container-engine-lib/lib/backend_impls/kubernetes/kubernetes_kurtosis_backend/engine_functions/create_engine.go

```diff
@@ -165,13 +179,34 @@ func CreateEngine(
 		}
 	}()
 
+	engineIngress, err := createEngineIngress(
+		ctx,
+		namespaceName,
+		engineAttributesProvider,
+		privateRESTAPIPortSpec,
+		kubernetesManager,
+	)
+	if err != nil {
+		return nil, stacktrace.Propagate(err, "An error occurred creating the engine ingress")
+	}
+	var shouldRemoveIngress = true
```

Comment:
why this is a `var` and why this always `true`? It seems I'm missing something in the behavior of of this code

Reply:
It is set to true here so the code tries to remove the ingress if the `CreateEngine` function returns early with an error.  This pattern of cleanup in case of error is used a lot at Kurtosis.  This variable is set to false at the end of the `CreateEngine` function if everything went well.